src/background/settings.js

```diff
@@ -4,9 +4,13 @@
  * @typedef Settings
  * @prop {string} apiUrl
  * @prop {string} buildType
+ * @prop {{ dsn: string, release: string}} [raven]
  * @prop {string} serviceUrl
  */
 
+// @ts-ignore - Ignore missing module if file has not been generated.
```

Comment:
The JSON file is always present. The reason that this raise a warning is that TS doesn't understand importing JSON by default. It can be unable with:

```
"resolveJsonModule": true
```

I think it is worth considering adding this option.

One nice thing about that is that the types are defined automatically:

![Screenshot 2021-11-04 at 13 19 29](https://user-images.githubusercontent.com/8555781/140315734-883c6f64-7b2c-4b01-908d-697af909635f.png)

Reply:
> The JSON file is always present. 

That's not true if the build hasn't run yet.

Thanks for the tip about `resolveJsonModule`. I have enabled it and modified the `make lint` command so it has a dependency on `build/settings.json` and also added a note for developers about why the file may see an error in their editor.